src/middlewared/middlewared/plugins/filesystem_/acl_linux.py

```diff
@@ -0,0 +1,573 @@
+
+        # linux output adds extra line to output if it's an absolute path and extra newline at end.
+        entries = gfacl.stdout.decode().splitlines()
+        entries = entries[:-1]
+
+        for entry in entries:
+            if entry.startswith("#"):
+                continue
+            ace = {
+                "default": False,
+                "tag": None,
+                "id": -1,
+                "perms": {
+                    "READ": False,
+                    "WRITE": False,
+                    "EXECUTE": False,
+                }
+            }
+
+            tag, id, perms = entry.rsplit(":", 2)
+            ace['perms'].update({
+                "READ": perms[0].casefold() == "r",
+                "WRITE": perms[1].casefold() == "w",
+                "EXECUTE": perms[2].casefold() == "x",
+            })
+            if tag.startswith('default'):
+                ace['default'] = True
+                tag = tag[8:]
+
+            ace['tag'] = tag.upper()
+            if id.isdigit():
+                ace['id'] = int(id)
+            ret['acl'].append(ace)
+
+        return ret
+
+    @accepts(
+        Str('path'),
+        Bool('simplified', default=True),
+    )
+    def getacl(self, path, simplified=True):
+        """
+        Return ACL of a given path. This may return a POSIX1e ACL or a NFSv4 ACL. The acl type is indicated
+        by the `ACLType` key.
+
+        Errata about ACLType NFSv4:
+
+        `simplified` returns a shortened form of the ACL permset and flags.
+
+        `TRAVERSE` sufficient rights to traverse a directory, but not read contents.
+
+        `READ` sufficient rights to traverse a directory, and read file contents.
+
+        `MODIFIY` sufficient rights to traverse, read, write, and modify a file. Equivalent to modify_set.
+
+        `FULL_CONTROL` all permissions.
+
+        If the permisssions do not fit within one of the pre-defined simplified permissions types, then
+        the full ACL entry will be returned.
+
+        In all cases we replace USER_OBJ, GROUP_OBJ, and EVERYONE with owner@, group@, everyone@ for
+        consistency with getfacl and setfacl. If one of aforementioned special tags is used, 'id' must
+        be set to None.
+
+        An inheriting empty everyone@ ACE is appended to non-trivial ACLs in order to enforce Windows
+        expectations regarding permissions inheritance. This entry is removed from NT ACL returned
+        to SMB clients when 'ixnas' samba VFS module is enabled. We also remove it here to avoid confusion.
+        """
+        if not os.path.exists(path):
+            raise CallError('Path not found.', errno.ENOENT)
+
+        return self.getacl_posix1e(path, simplified)
+
+    @private
+    def setacl_nfs4(self, job, data):
+        raise CallError('NFSv4 ACLs are not yet implemented.', errno.EOPNOTSUP)
+
+    @private
+    def setacl_posix1e(self, job, data):
+        job.set_progress(0, 'Preparing to set acl.')
+        if osc.IS_FREEBSD:
```

Comment:
I think that cannot ever be true, as _linux.py is not loaded on bsd

Reply:
Yeah. I'll go through and clean up the osc stuff from both sides.